File: code_combat/7_Sarven_Desert/294-Mushroom_Noise/mushroomNoise.js
// Defeat the skeleton and open the chest.

function onSpawn (event) {
    // Pet should find the health potion (type is "potion"): 
    var potion = pet.findNearestByType("potion");
    // and fetch it:
    if (potion) pet.fetch(potion);
    // Pet should find the gold key (type is "gold-key"):
    var goldKey = pet.findNearestByType("gold-key");
    // and fetch it:
    if (goldKey) pet.fetch(goldKey);
}

// Pet can find more than just items:
var skeleton = pet.findNearestByType("skeleton");
pet.on("spawn", onSpawn);

while(true) {
    if (skeleton.health > 0 && !hero.hasEffect("hide"))
        hero.attack(skeleton);
    else if (skeleton.health <= 0)
        hero.moveXY(31, 38);
}

File: code_combat/7_Sarven_Desert/294-Mushroom_Noise/mushroom_noise.py
def onSpawn(event):
    potion = pet.findNearestByType("potion")
    if potion:
        pet.fetch(potion)
    goldKey = pet.findNearestByType("gold-key")
    if goldKey:
        pet.fetch(goldKey)


skeleton = pet.findNearestByType("skeleton")
pet.on("spawn", onSpawn)

while True:
    if skeleton.health > 0:
        hero.attack(skeleton)
    else:
        hero.moveXY(31, 38)
